addons/pkg/constants/constants.go

```diff
@@ -282,6 +282,9 @@ const (
 	// logic
 	AddCBMissingFieldsAnnotationKey = "tkg.tanzu.vmware.com/add-missing-fields-from-tkr"
 
+	// UseCustomCNIProvider is the label for clusters that use a custom CNI provider
+	UseCustomCNIProvider = "tkg.tanzu.vmware.com/use-custom-cni-provider"
```

Comment:
1. In validation webhook check if spec.cni is nil if this annotation is given
2. In defaulting webhook skip adding spec.cni from CBT if this annotation is given
3. Fix validating webhooks where spec.cni is mandatory.

Reply:
checklist done